### PR TITLE
Add bloom filter dictionary.

### LINKED ISSUE #80: Add BloomFilter dictionary
Bloom filters provide a space efficient data structure for large dictionaries.
The cost for this space savings is the possibility of false positives.
While the false positive probability can be defined, it cannot be eliminated without also eliminating the space savings.

Provide a BloomFilterDictionary implementation.

### PR DESCRIPTION
**Pull request by dfish3r**
Adds guava as an optional dependency for it's BloomFilter impl.
Fixes #80.

### CODE REVIEW

File: src/main/java/org/passay/dictionary/BloomFilterDictionary.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+public class BloomFilterDictionary implements Dictionary
+{
+
+  /** Filter used for searching. */
+  private final BloomFilter<String> bloomFilter;
+
+
+  /**
+   * Creates a new dictionary instance from the supplied {@link BloomFilter}.
```
Comment (serac) [inline src/main/java/org/passay/dictionary/BloomFilterDictionary.java]: You should consider making a recommendation here or in the class javadoc for the false-positive probability, or at least discuss tradeoffs. I'm concerned that the default Guava value of 3% is too high.

Comment (dfish3r) [reply src/main/java/org/passay/dictionary/BloomFilterDictionary.java]: I'm not sure what I'd say. I've tried to make it clear that bloom filters will operate differently than the other dictionaries. But advice about a particular fpp seems very subjective.

Comment (serac) [reply src/main/java/org/passay/dictionary/BloomFilterDictionary.java]: In the spirit of being descriptive over prescriptive, I'll take a stab at a helpful note to be added to what you have for the constructor:

> The _fpp_ (false-positive probability) parameter of the given Bloom filter is a vitally important configuration concern. If it is too high, one risks user frustration due to rejection of valid passwords; if it is too low, one risks excessive storage costs. Finding the proper balance between acceptable user experience and storage costs is worth the time and effort required in testing. The Guava default value of 3% is likely unsuitable for many if not most deployments.

Comment (dfish3r) [reply src/main/java/org/passay/dictionary/BloomFilterDictionary.java]: Javadocs updated.

**Pull request merged**